src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -60,18 +97,33 @@ private Pattern compileAllowlist(List<String> allowlist) {
         return Pattern.compile(joined);
     }
 
+    /**
+     * Gets the listener.
+     *
+     * @return the listener.
+     */
+    public String listener() {
+        return listener.toString();
+    }
+
     @Override
-    public String toString() {
+     public String toString() {
         return "PrometheusMetricsReporterConfig{" +
                 "allowlist=" + allowlist +
-                ", port=" + port +
+                ", listener=" + listener +
                 '}';
     }
 
+    /**
+     * Start the HTTP server for exposing metrics.
+     *
+     * @return An optional HTTPServer instance if started successfully, otherwise empty.
+     */
     public synchronized Optional<HTTPServer> startHttpServer() {
         try {
-            HTTPServer httpServer = new HTTPServer(port, true);
-            LOG.info("HTTP server started on port " + port);
+            InetSocketAddress address = listener.getInetSocketAddress();
+            HTTPServer httpServer = new HTTPServer(address.getPort(), true);
```

Comment:
```suggestion
                        HTTPServer httpServer = new HTTPServer(address, CollectorRegistry.defaultRegistry, true);

```

Reply:
Considering the discussion above, I think we should just call `new HTTPServer(host, port, true)`

Reply:
Ok